src/lib/llm/openai.py

```
@@ -22,17 +22,7 @@ def query(self, prompt: str) -> str:
-            if "output" in response_data and len(response_data["output"]) > 0:
-                output = response_data["output"][0]
-                if "content" in output and len(output["content"]) > 0:
-                    content_item = output["content"][0]
-                    if content_item["type"] == "output_text":
-                        return content_item["text"]
-
-            raise OpenAIError("No valid response from OpenAI")
-
+            return self._extract_text_response(response.json())
         except Exception as e:
             raise OpenAIError(f"Error querying OpenAI: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)